core/src/commonMain/kotlin/com/powersync/sync/SyncStream.kt

```diff
@@ -172,7 +190,7 @@ internal class SyncStream(
                 contentType(ContentType.Application.Json)
                 headers {
                     append(HttpHeaders.Authorization, "Token ${credentials.token}")
-                    append("User-Id", credentials.userId ?: "")
+                    append("User-Agent", powerSyncUserAgent())
```

Comment:
Given that we're doing this on all requests, a better place to put this may be in `init`:

```Kotlin
fun HttpClientConfig<*>.configureClient() {
    install(HttpTimeout)
    install(ContentNegotiation)

    defaultRequest {
        headers {
            append("User-Agent", $LIBRARY_NAME/$LIBRARY_VERSION ${getOS()}")
        }
    }
}
```